prboom2/src/gl_main.c

```diff
@@ -2328,20 +2328,29 @@ static void gld_DrawSprite(GLSprite *sprite)
 
 static void gld_AddHealthBar(mobj_t* thing, GLSprite *sprite)
 {
-  if (((thing->flags & (MF_COUNTKILL | MF_CORPSE)) == MF_COUNTKILL) && (thing->health > 0))
+  if ((thing->flags & MF_SHOOTABLE) && thing->info->spawnhealth > 0 &&
+      thing->health > 0)
   {
     GLHealthBar hbar;
     int health_percent = thing->health * 100 / thing->info->spawnhealth;
 
-    hbar.cm = -1;
-    if (health_percent <= health_bar_red)
-      hbar.cm = CR_RED;
-    else if (health_percent <= health_bar_yellow)
-      hbar.cm = CR_YELLOW;
-    else if (health_percent <= health_bar_green)
-      hbar.cm = CR_GREEN;
+    hbar.r = hbar.g = -1.0f;
+    if (health_percent <= 0) {
+      hbar.r = 1.0f;
+      hbar.g = 0.0f;
+    } else if (health_percent <= health_bar_red) {
+      hbar.r = 1.0f;
+      hbar.g = (double)health_percent / (health_bar_yellow - health_bar_red);
+    } else if (health_percent <= health_bar_yellow) {
+      hbar.r = (double)(health_bar_yellow - health_percent) /
+               (health_bar_yellow - health_bar_red);
+      hbar.g = 1.0f;
+    } else if (health_percent <= health_bar_green) {
```

Comment:
If this applies to all shootable objects, it applies to other players as well, who might have more health than `spawnhealth`.

Reply:
I can't see anything negative stemming from a mobj having health_percent>100.